config/rollup.config.js

```diff
@@ -0,0 +1,212 @@
+if (pkgInfo.dependencies) external.push(...Object.keys(pkgInfo.dependencies));
+
+// TODO: the packages do not have an ESM build, so we'll just bundle and ship it
+external = external.filter(
+  x => x !== 'query-string' && x !== 'styled-components'
+);
+
+const externalPredicate = new RegExp(`^(${external.join('|')})($|/)`);
+const externalTest = id => {
+  if (id === 'babel-plugin-transform-async-to-promises/helpers') {
+    return false;
+  }
+
+  return externalPredicate.test(id);
+};
+
+const terserMinified = terser({
+  sourcemap: true,
+  warnings: true,
+  ecma: 5,
+  ie8: false,
+  toplevel: true,
+  compress: {
+    keep_infinity: true,
+    pure_getters: true,
+    passes: 10
+  },
+  output: {
+    comments: false
+  }
+});
+
+const makePlugins = () =>
+  [
+    nodeResolve({
+      mainFields: ['module', 'jsnext', 'main'],
+      browser: true
+    }),
+    commonjs({
+      ignoreGlobal: true,
+      include: /\/node_modules\//,
+      namedExports: {
+        react: Object.keys(require('react')),
+        'react-is': Object.keys(require('react-is'))
+      }
+    }),
+    buble({
+      transforms: {
+        unicodeRegExp: false,
+        dangerousForOf: true,
+        dangerousTaggedTemplateString: true
+      },
+      objectAssign: 'Object.assign',
+      exclude: 'node_modules/**'
+    }),
+    babel({
+      babelrc: false,
+      presets: [],
+      plugins: [
+        '@babel/plugin-transform-object-assign',
+        [
+          '@babel/plugin-transform-react-jsx',
+          {
+            pragma: 'React.createElement',
+            pragmaFrag: 'React.Fragment',
+            useBuiltIns: true
+          }
+        ],
+        [
+          'babel-plugin-transform-async-to-promises',
+          {
+            inlineHelpers: true,
+            externalHelpers: true
+          }
+        ],
+        [
+          'babel-plugin-transform-react-remove-prop-types',
+          {
+            mode: isProduction ? 'remove' : 'wrap',
+            removeImport: isProduction
+          }
+        ]
+      ].filter(Boolean)
+    }),
+    isProduction &&
+      replace({
+        'process.env.NODE_ENV': JSON.stringify('production')
+      }),
+    isProduction && terserMinified
+  ].filter(Boolean);
+
+const config = {
+  input: pkgInfo.source || './src/index.js',
+  external: externalTest,
+  treeshake: {
+    propertyReadSideEffects: false
+  }
+};
+
+const globals = {
+  react: 'React',
+  'react-is': 'ReactIs',
+  'react-dom': 'ReactDOM',
+  'prop-types': 'PropTypes'
+};
+
+export default [
+  {
+    ...config,
+    plugins: makePlugins(false),
+    output: [
+      {
+        sourcemap: true,
+        legacy: true,
+        freeze: false,
+        esModule: false,
+        file: `./dist/${name}.js`,
+        format: 'cjs'
+      }
+    ]
+  },
+  {
+    ...config,
+    external: id => externalTest(id),
+    plugins: [...makePlugins(), !isProduction && unpkg()].filter(Boolean),
```

Comment:
Is there a big downside to including the unpkg URLs in the production build? I've omitted them here but i'm not totally convinced either way.

Reply:
well, so I wouldn't add `isProduction` via `NODE_ENV` in the first place. This is a build setup that should be repeatable in any circumstance, so when we're publishing all these bundles we don't want to deal with running `env NODE_ENV=production yarn build && yarn build` and then maintain different names here all depending on an env flag.

So I'd just remove it and make sure that this config always builds the same set of files, oftentimes production (minified) and unminified bundles.

I think in the case of Spectacle it'd also make sense to just remove the production modes / bundles. There's no real use of `process.env.NODE_ENV` in the code yet and there's no need to actually make Spectacle as small as possible